pallets/author-slot-filter/src/lib.rs

```diff
@@ -128,20 +142,26 @@ pub mod pallet {
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
-		/// Update the eligible ratio. Intended to be called by governance.
+		/// Update the eligible count. Intended to be called by governance.
 		#[pallet::weight(T::WeightInfo::set_eligible())]
-		pub fn set_eligible(origin: OriginFor<T>, new: Percent) -> DispatchResultWithPostInfo {
+		pub fn set_eligible(
+			origin: OriginFor<T>,
+			new: EligibilityValue,
+		) -> DispatchResultWithPostInfo {
 			ensure_root(origin)?;
-			EligibleRatio::<T>::put(&new);
+			EligibleCount::<T>::put(&new);
 			<Pallet<T>>::deposit_event(Event::EligibleUpdated(new));
 
 			Ok(Default::default())
 		}
 	}
 
-	/// The percentage of active authors that will be eligible at each height.
+	/// The type of eligibility to use
+	pub type EligibilityValue = Option<NonZeroU32>;
```

Comment:
why Option ?

Reply:
I kept the same interface as provided by https://docs.substrate.io/rustdocs/latest/sp_std/num/struct.NonZeroU32.html#method.new to make it easier to port once I file a couple of upstream PRs to substrate. `new` checks if the value is `> 0` putting it in `Some(...)`, else returning a `None`.

Reply:
I not talking about the new signature, but about the storage type, the storage should not contains an Option, it's useless because all storage values are already optional, and you already provided a default value in case the storage item is empty.

Reply:
So please remove this Option from the type used for storage.